File: src/AutoRest.SdkExplorer/Model/Azure/AzureSubscription.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

#nullable disable

using System.Collections.Generic;

namespace AutoRest.SdkExplorer.Model.Azure
{
    public class AzureSubscription
    {
        public List<AzureResourceGroup> ResourceGroups { get; set; } = new List<AzureResourceGroup>();
        public List<AzureResource> Resources { get; set; } = new List<AzureResource>();
        public string SubscriptionId {get;set;}
        public string Name {get;set;}
        public string Id {get;set;}
        public bool IsDefault { get; set; }

        public AzureSubscription()
        {
        }
    }
}

File: src/AutoRest.SdkExplorer.TypeScript/src/Model/Azure/AzureSubscription.ts
import { AzureResource } from "./AzureResource";
import { AzureResourceGroup } from "./AzureResourceGroup";

export class AzureSubscription implements AutoRest.SdkExplorer.Interface.AzureSubscription {
    resourceGroups: AzureResourceGroup[];
    resources: AzureResource[];
    subscriptionId: string;
    name: string;
    id: string;
    isDefault: boolean;

    constructor(data: AutoRest.SdkExplorer.Interface.AzureSubscription) {
        this.resourceGroups = data.resourceGroups?.map(rg => new AzureResourceGroup(rg)) ?? [];
        this.resources = data.resources?.map(r => new AzureResource(r)) ?? [];
        this.subscriptionId = data.subscriptionId!;
        this.name = data.name!;
        this.id = data.id!;
        this.isDefault = data.isDefault ?? false;
    }
}
